### PR TITLE
MWPW-135950 Add "commerce" library to Milo deps, update commerce, merch and OST blocks

### PR DESCRIPTION
**Pull request by vladen**
## Description
This PR updates `commerce`, `merch` and `OST` blocks to consume the latest version of [wcms/commerce](https://git.corp.adobe.com/wcms/tacocat.js/tree/develop/packages/tacocat) library bundle (former `tacocat.js`). The bundle is added to Milo dependencies and imported from there to benefit from same domain origin.

This PR should be only merged after commerce release 2 demo!

Resolves: [MWPW-135950](https://jira.corp.adobe.com/browse/MWPW-135950)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/ilyas/merch?martech=off&georouting=off
- After: https://mwpw-135950--milo--vladen.hlx.page/drafts/ilyas/merch?martech=off&georouting=off

Note: `modHeader` or similar browser extension is needed to workaround CORS errors reported for `wcs.adobe.com`

### CODE REVIEW

File: test/blocks/merch/mocks/fetch.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+export async function mockFetch() {
+  // this path allows to import this mock from tests for other blocks (e.g. commerce)
+  const literals = JSON.parse(await readFile({ path: '../merch/mocks/literals.json' }));
+  const offers = JSON.parse(await readFile({ path: '../merch/mocks/offers.json' }));
+
+  const { fetch } = window;
+  sinon.stub(window, 'fetch').callsFake((...args) => {
+    const { href, pathname, searchParams } = new URL(String(args[0]));
+    // literals mock
+    if (href === 'https://milo.adobe.com/drafts/vlassenko/price-literals.json') {
```
Comment (yesil) [inline test/blocks/merch/mocks/fetch.js:13]: literals to move to a final folder from drafts

File: libs/blocks/ost/ost.js [hunk 2]
```diff
@@ -1,36 +1,40 @@
...
-import { ENV_PROD, getTacocatEnv } from '../merch/merch.js';
 
+export const AOS_API_KEY = 'wcms-commerce-ims-user-prod';
+export const CHECKOUT_CLIENT_ID = 'creative';
 const IMS_COMMERCE_CLIENT_ID = 'aos_milo_commerce';
+const IMS_SCOPE = 'AdobeID,openid';
+const IMS_ENV = 'prod';
 const IMS_PROD_URL = 'https://auth.services.adobe.com/imslib/imslib.min.js';
-const OST_VERSION = '1.14.0';
+const OST_VERSION = '1.11.0';
```
[the remark this answers is not in the document]
Comment (vladen) [reply libs/blocks/ost/ost.js:10]: Should be a merge leftover, restoring

File: libs/blocks/merch/merch.js [hunk 3]
```diff
@@ -1,240 +1,132 @@
...
-  if (fail) {
-    return undefined;
+export async function getPriceContext(el, params) {
+  const context = await getCommerceContext(el, params);
+  if (!context) return null;
+  const displayOldPrice = context.promotionCode ? params.get('old') : undefined;
+  const displayPerUnit = params.get('seat');
+  const displayRecurrence = params.get('term');
+  const displayTax = params.get('tax');
+  const type = params.get('type');
```
[the remark this answers is not in the document]
Comment (vladen) [reply libs/blocks/merch/merch.js:60]: Let's support it in a future PR to constrain this large one.

File: libs/blocks/ost/ost.js [hunk 4]
```diff
@@ -1,36 +1,40 @@
...
 const OST_BASE = `https://www.stage.adobe.com/special/tacocat/ost/lib/${OST_VERSION}`;
 const OST_SCRIPT_URL = `${OST_BASE}/index.js`;
 const OST_STYLE_URL = `${OST_BASE}/index.css`;
-
-export const AOS_API_KEY = 'wcms-commerce-ims-user-prod';
-export const CHECKOUT_CLIENT_ID = 'creative';
+/** @see https://git.corp.adobe.com/PandoraUI/core/blob/master/packages/react-env-provider/src/component.tsx#L49 */
+export const PANDORA_ENV = 'production';
```
[the remark this answers is not in the document]
Comment (vladen) [reply libs/blocks/ost/ost.js:15]: Sounds good

**Review by yesil — COMMENTED** [review]
Please update path of the literals to a final location.

File: libs/blocks/merch/merch.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (npeltier) [reply libs/blocks/merch/merch.js:60]: yes @yesil i'll take care of it :)

**Comment by vladen** [comment]
> Please update path of the literals to a final location.

Done

File: libs/blocks/merch/merch.js [hunk 5]
```diff
@@ -1,243 +1,138 @@
...
+    cta.classList.add('button-l');
   }
-  const { searchParams } = new URL(el.href);
-  const osi = searchParams.get('osi');
-  const type = searchParams.get('type');
-  if (!(osi && type)) {
-    el.textContent = '';
-    el.setAttribute('aria-details', 'Invalid merch block');
-    return undefined;
+  if (el.firstElementChild?.tagName === 'STRONG' || el.parentElement?.tagName === 'STRONG') {
```
Comment (Axelcureno) [inline libs/blocks/merch/merch.js:88]: Let's consider maybe delegating this logic to `decorateButtons()` in [decorate.js].(https://github.com/adobecom/milo/blob/main/libs/utils/decorate.js)

File: libs/blocks/merch/merch.js [hunk 6]
```diff
@@ -1,243 +1,138 @@
...
-  const checkoutWorkflowStep = searchParams
+export async function getCheckoutContext(el, params) {
+  const context = await getCommerceContext(el, params);
+  if (!context) return null;
+  const { settings } = await initService();
+  const { checkoutClientId } = settings;
+  const checkoutMarketSegment = params.get('marketSegment');
+  const checkoutWorkflow = params.get('workflow')
+    ?? getMetadata('checkout-workflow')
+    ?? settings.checkoutWorkflow;
```
Comment (VKniaz) [inline libs/blocks/merch/merch.js:43]: can be changed to settings?.checkoutWorkflow

File: libs/blocks/merch/merch.js [hunk 7]
```diff
@@ -1,243 +1,138 @@
...
+    ...getConfig(),
+    commerce: { priceLiteralsURL },
+  }));
 }
 
-function isCTA(type) {
-  return type === 'checkoutUrl';
+export async function getCommerceContext(el, params) {
+  const wcsOsi = params.get('osi');
+  if (!wcsOsi) return null;
```
Comment (VKniaz) [inline libs/blocks/merch/merch.js:16]: Maybe we should have a descriptive error or log for that.

File: libs/blocks/commerce/commerce.js [hunk 8]
```diff
@@ -125,49 +89,43 @@ export const decorateOfferDetails = async (el, of, searchParams) => {
...
   el.append(offerDetailsList);
-};
+}
 
-export const handleSearch = async (event, el) => {
-  const displaySearchError = () => {
-    const notValidUrl = createTag('h4', { class: 'not-valid-url' }, 'Not a valid offer link');
-    el.append(notValidUrl);
-  };
+export async function handleSearch(event, el) {
```
Comment (VKniaz) [inline libs/blocks/commerce/commerce.js:94]: Maybe rename it to a more specific handleOfferSearch

**Review by VKniaz — APPROVED** [review]

File: libs/blocks/merch/merch.js (hunk 6 above)
Comment (vladen) [reply libs/blocks/merch/merch.js:43]: No, per commerce library design, `settings. checkoutWorkflow` will be always set to a configured or default value (UCv3).

File: libs/blocks/merch/merch.js (hunk 5 above)
Comment (vladen) [reply libs/blocks/merch/merch.js:88]: @Axelcureno I tried to use `decorateButtons` but apparently in merch block we have slightly differing (simpler) logic. So far, `decorateButtons` can help in adding the `blue` CSS class to merch CTA, but not `button-l` and `con-button` classes for all the cases on the test page ("after" link in PR description).
I propose to keep this as is for now and discuss later, as more CTA variants come.

File: libs/blocks/commerce/commerce.js (hunk 8 above)
Comment (vladen) [reply libs/blocks/commerce/commerce.js:94]: Renamed, thanks.

File: libs/blocks/merch/merch.js [hunk 9]
```diff
@@ -1,243 +1,138 @@
...
-  });
-  el.replaceWith(price);
-  return price;
+export default async function init(el) {
+  if (!el?.classList?.contains('merch')) return undefined;
+  await polyfill();
+  const { searchParams: params } = new URL(el.href);
+  const service = await initService();
+  const log = service.log.module('merch');
+  const build = params.get('type') === 'checkoutUrl' ? buildCta : buildPrice;
```
Comment (yesil) [inline libs/blocks/merch/merch.js:127]: variable name `build` is confusing.  Can we give an explicit name such as `merchElement` or `merchEl` or just `el`.

File: libs/blocks/merch/merch.js (hunk 5 above)
Comment (yesil) [reply libs/blocks/merch/merch.js:88]: actually `button-l` and `con-button` were our initial assumptions but maybe we should simply use decorateButtons

**Comment by vladen** [comment]
@yesil please check my answer to Axel's suggestion about using `decorateButtons`

**Review by yesil — APPROVED** [review]

**Review by Axelcureno — APPROVED** [review]

**Review by honstar — APPROVED** [review]
LGTM

**Comment by vladen** [comment]
@honstar tests are fixed, all checks are green again, thanks

**Pull request merged**